Sources/PaylikeSDK/Views/ViewModel/PayButtonViewModel.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  PayButtonViewModel.swift
+//  
+//
+//  Created by Székely Károly on 2023. 05. 26..
+//
+
+import Foundation
+import SwiftUI
+import PaylikeClient
+
+public let defaultButtonStyle = PayButtonStyle()
+
+public class PayButtonViewModel: ObservableObject {
+    @Published var amount: PaymentAmount
+    @Published var styling: PayButtonStyle
+    private var submit: () async -> Void
```

Comment:
The submit function is stored privately and can be set through the init. We can call this func with the action public function, which is called in the Paybutton. Why is this wrapping? Isn't creating a public get, private set property or field enough?

Reply:
It is most likely enough, i was just too eager, and didn't really forsee how the button will call this method beforehand